macOS install fails under Homebrew 6.0 tap trust: "Refusing to load cask ... from untrusted tap"

## Summary

Since macOS GitHub runner images picked up **Homebrew 6.0** (released 2026-06-11, which introduced [tap trust](https://docs.brew.sh/Tap-Trust)), the macOS/darwin install path fails for the **production** Senzing SDK cask. This breaks CI in every consuming repo that runs `production-v4` on `macos-latest` (e.g. `senzing-garage/sz-python-tools`).

## Error

```
[INFO] brew install --cask senzingsdk (latest)
##[error]Refusing to load cask senzing/senzingsdk/senzingsdk from untrusted tap senzing/senzingsdk.
Run `brew trust --cask senzing/senzingsdk/senzingsdk` or `brew trust senzing/senzingsdk` to trust it.
##[error]Process completed with exit code 1.
```

## Root cause

`darwin/install-senzing.sh` does:

```bash
brew tap "$PRODUCTION_TAP"          # Senzing/senzingsdk
brew tap "$STAGING_TAP"             # senzing-factory/senzingsdk-staging
brew install --cask "$cask"
```

Homebrew's official `cask_tap_migrations` redirect the short cask name `senzingsdk` to the third-party `senzing/senzingsdk` tap. Under Homebrew 6.0, an auto-redirected cask from an untrusted third-party tap is **refused by default** — even when `HOMEBREW_REQUIRE_TAP_TRUST` is not set. The staging cask (`senzingsdk-staging`) has no such migration entry, so it currently only emits an advisory and still installs.

## Suggested fix

After tapping, explicitly trust the cask(s) the action installs, e.g.:

```bash
brew tap "$PRODUCTION_TAP"
brew trust --cask "$PRODUCTION_TAP/$cask"   # senzing/senzingsdk/senzingsdk
brew install --cask "$cask"
```

(`brew trust` was added in Homebrew 6.0; the action could guard on `brew trust` existing for older Homebrew.) Trusting the specific cask is preferred over trusting the whole tap or disabling enforcement globally.

## Workaround in consuming repos (stopgap)

Setting `HOMEBREW_NO_REQUIRE_TAP_TRUST=1` in the workflow env lets `brew install --cask` succeed during the transition — but that env var is deprecated and Homebrew says it "will be removed in a later release," so the durable fix belongs here in the action. Example: senzing-garage/sz-python-tools#412.

## Environment

- Action: `senzing-factory/github-action-install-senzing-sdk@v5` (5.0.3)
- Runner: `macos-latest` (macOS 15 / arm64), Homebrew 6.0
- First observed: ~2026-06-24 (matches the runner image rolling out Homebrew 6.0)

--- a/.github/workflows/test-install-scripts.yaml
+++ b/.github/workflows/test-install-scripts.yaml
@@ -127,7 +127,13 @@ jobs:
           SENZING_INSTALL_VERSION: ${{ matrix.senzingsdk-version }}
           DARWIN_INSTALLER: ${{ matrix.darwin-installer }}
           SENZINGSDK_TOKEN: ${{ steps.app-token.outputs.token }}
-        run: bash darwin/install-senzing.sh
+          # Force Homebrew 6.0 tap-trust enforcement so the homebrew rows keep
+          # exercising the fully-qualified-install trust fix even if a future
+          # runner image changes the default. No-op on the native rows / older brew.
+          HOMEBREW_REQUIRE_TAP_TRUST: 1
+        run: |
+          echo "[INFO] $(brew --version | head -1)"
+          bash darwin/install-senzing.sh
 
   test-windows:
     name: "Windows: ${{ matrix.senzingsdk-version }} (${{ matrix.windows-installer || 'auto' }})"

--- a/darwin/install-senzing.sh
+++ b/darwin/install-senzing.sh
@@ -321,6 +321,31 @@ determine-homebrew-version() {
 }
 
 ############################################
+# homebrew-cask-ref
+# Echoes the fully-qualified "<tap>/<cask>" reference for the current
+# REPO_KIND. Installing by this fully-qualified name is an explicit request
+# for that exact cask, so Homebrew 6.0 trusts just that item — instead of
+# refusing the auto-migrated bare short name `senzingsdk`, which redirects
+# (via Homebrew's cask_tap_migrations) into the untrusted third-party tap.
+# GLOBALS:
+#   REPO_KIND
+#   PRODUCTION_TAP / PRODUCTION_CASK
+#   STAGING_TAP / STAGING_CASK
+############################################
+homebrew-cask-ref() {
+
+  case "$REPO_KIND" in
+    production) echo "$PRODUCTION_TAP/$PRODUCTION_CASK" ;;
+    staging) echo "$STAGING_TAP/$STAGING_CASK" ;;
+    *)
+      echo "[ERROR] unsupported repository '$REPO_KIND' for homebrew install" >&2
+      return 1
+      ;;
+  esac
+
+}
+
+############################################
 # install-via-homebrew
 # GLOBALS:
 #   REPO_KIND
@@ -329,15 +354,12 @@ determine-homebrew-version() {
 ############################################
 install-via-homebrew() {
 
-  local cask
   case "$REPO_KIND" in
     production)
-      cask="$PRODUCTION_CASK"
       echo "[INFO] brew tap $PRODUCTION_TAP"
       brew tap "$PRODUCTION_TAP"
       ;;
     staging)
-      cask="$STAGING_CASK"
       if [ -z "$SENZINGSDK_TOKEN" ]; then
         echo "[ERROR] senzingsdk-token is required for homebrew installs from the staging tap (private repo $STAGING_TAP_REPO)"
         exit 1
@@ -351,14 +373,18 @@ install-via-homebrew() {
       ;;
   esac
 
+  # Fully-qualified ref (tap/cask) so Homebrew 6.0 tap-trust accepts it.
+  local cask_ref
+  cask_ref="$(homebrew-cask-ref)"
+
   export HOMEBREW_SENZING_ACCEPT_EULA="i_accept_the_senzing_eula"
   if [ -n "$HOMEBREW_PIN_VERSION" ]; then
     export HOMEBREW_SENZING_SDK_VERSION="$HOMEBREW_PIN_VERSION"
-    echo "[INFO] brew install --cask $cask (pinned to $HOMEBREW_PIN_VERSION)"
+    echo "[INFO] brew install --cask $cask_ref (pinned to $HOMEBREW_PIN_VERSION)"
   else
-    echo "[INFO] brew install --cask $cask (latest)"
+    echo "[INFO] brew install --cask $cask_ref (latest)"
   fi
-  brew install --cask "$cask"
+  brew install --cask "$cask_ref"
 
   link-homebrew-prefix
   publish-homebrew-env

--- a/tests/install-senzing-darwin.bats
+++ b/tests/install-senzing-darwin.bats
@@ -190,3 +190,32 @@ setup() {
   determine-build-for-version
   [ "$SENZINGSDK_BUILD_URL" = "https://example.com/senzingsdk_5.0.0.12345.pkg" ]
 }
+
+# ---------------------------------------------------------------------------
+# homebrew-cask-ref: fully-qualified tap/cask name (Homebrew 6.0 tap trust)
+# ---------------------------------------------------------------------------
+
+@test "homebrew-cask-ref: production -> fully-qualified prod cask" {
+  PRODUCTION_TAP="Senzing/senzingsdk"
+  PRODUCTION_CASK="senzingsdk"
+  REPO_KIND="production"
+  run homebrew-cask-ref
+  [ "$status" -eq 0 ]
+  [ "$output" = "Senzing/senzingsdk/senzingsdk" ]
+}
+
+@test "homebrew-cask-ref: staging -> fully-qualified staging cask" {
+  STAGING_TAP="senzing-factory/senzingsdk-staging"
+  STAGING_CASK="senzingsdk-staging"
+  REPO_KIND="staging"
+  run homebrew-cask-ref
+  [ "$status" -eq 0 ]
+  [ "$output" = "senzing-factory/senzingsdk-staging/senzingsdk-staging" ]
+}
+
+@test "homebrew-cask-ref: unknown repo kind errors" {
+  REPO_KIND="custom"
+  run homebrew-cask-ref
+  [ "$status" -ne 0 ]
+  [[ "$output" =~ "unsupported repository" ]]
+}

